libsplinter/src/node_id/store/error.rs

```diff
@@ -43,3 +44,16 @@ impl Error for NodeIdStoreError {
         }
     }
 }
+
+impl From<diesel::result::Error> for NodeIdStoreError {
+    fn from(err: diesel::result::Error) -> Self {
+        Self::InternalError(InternalError::from_source(Box::new(err)))
+    }
+}
+
+#[cfg(feature = "diesel")]
+impl From<diesel::r2d2::PoolError> for NodeIdStoreError {
+    fn from(err: diesel::r2d2::PoolError) -> Self {
+        Self::InternalError(InternalError::from_source(Box::new(err)))
```

Comment:
Instead of InternalError, top level ResourceTemporarilyUnavailableError should be added to NodeIdStoreError and  this conversion should it set PoolError to ResourceTemporarilyUnavailableError